protocol/0087-VAMM-automated_market_maker.md

```diff
@@ -0,0 +1,177 @@
+
+Each party may have only one AMM configuration per market.
+
+## Process Overview
+
+The configuration and resultant lifecycle of an automated market maker is as follows:
+
+- Party funds a key which will be used by the strategy with desired token amounts.
+- Party submits a transaction containing configuration for the strategy on a given market. This will contain:
+  - Amount of funds to commit
+  - Price bounds (upper, lower, base)
+- Additionally, the transaction should contain data related to the setup of the position but which does not need to be stored:
+  - Maximum slippage (%), used for rebasing position when creating/updating AMM
+- Once accepted, the network will transfer funds to a sub-account and use the other parameters for maintaining the position.
+- At each block, the party's available balance (including margin and general accounts) for trading on the market will be checked. If the total balance is `0` the AMM configuration will be cancelled. 
+- If the party submits a `CancelAMM` transaction the AMM configuration for that party, on that market, will be cancelled. All active orders from the AMM will be cancelled and all funds and positions associated with the sub-account will be transferred back to the main account.
+
+## Sub-Account Configuration
+
+Each main Vega key will have one associated sub account for a given market, on which an AMM may be set up. The account key should be generated through a hash of the main account key plus the ID of the market to generate a valid Vega address in a predictable manner. Outside of the AMM framework the sub-accounts are treated identically to any other account, they will have the standard associated margin/general/bond accounts and be able to place orders if required as with any other account. The key differentiator is that no external party will have the private key to control these accounts directly. The maintenance of such an account will be performed through a few actions:
+
+- Creation: A sub-account will be funded when a user configures an AMM strategy with a set of criteria and a commitment amount. At this point in time the commitment amount will be transferred to the sub-account's general account and the AMM strategy will commence
+- Cancellation: When the AMM is cancelled all active orders are first cancelled. Following this cancellation, all funds in the sub-account's margin account should be transferred to the associated main account's margin account, with the same then happening for funds in the general account. Finally, any associated non-zero position on the market should be reassigned from the sub-account to the main account. At this point processing can continue, allowing the standard margining cycle to perform any required transfers from margin to general account.
```

Comment:
`Finally, any associated non-zero position on the market should be reassigned from the sub-account to the main account.` I'm not sure this is right, we should probably just try to close the position, if it's possible then the pool can be cancelled, if not we close as much of the position as we can, and leave the pull open.

Reply:
The challenge I see with that would be that a large LP could end up with a pretty substantial position compared to the market, which they'd have to either close all at once and eat a load of slippage or manually bring down their base position towards the current price over time, which is a bit of a pain.

Reply:
I certainly don't love the sound of "if the position can't be closed we leave the pool open and you can't exit".  

So I am leaning towards: if the user wishes to cancel their AMM then their position is dumped on their main key and the AMM is removed. 

The UI can offer to submit a market order for them to close the position (if there is other volume available).

Reply:
After discussing with @jeremyletang it sounds like this transfer of positions would be pretty painful to implement, so it may be easiest to say if you don't want to eat the slippage you just have size down bit-by-bit